dashboard/app/controllers/hoc_download_controller.rb

```diff
@@ -0,0 +1,42 @@
+class HocDownloadController < ApplicationController
+
+  DOWNLOAD_URL_BASE = '//downloads.code.org/hourofcode/'
+
+  def index
+    @product_name = params[:product]
+
+    case @product_name
+      when Script::MINECRAFT_NAME
+        @app_name = t('hoc_download.minecraft_name')
+        @file_prefix = 'MC'
+        @og_image_url = CDO.code_org_url('/images/mc/mc_social.jpg')
+      when Script::STARWARS_NAME
+        @app_name = t('hoc_download.starwars_blocks_name')
+        @file_prefix = 'StarWarsBlocks'
+        @og_image_url = CDO.code_org_url('/images/star-wars-announcement.jpg')
+
+        # for JavaScript version (Star Wars only)
+        @app_name_js = t('hoc_download.starwars_javascript_name')
+        @file_prefix_js = 'StarWarsJS'
+        @show_js_links = true
+      else
+        render_404
+        return
+    end
+
+    @hoc_url = CDO.code_org_url("/#{@product_name}")
+
+    @english_windows_url = DOWNLOAD_URL_BASE + @product_name + '/' + @file_prefix + "EnglishSetup.exe"
```

Comment:
It doesn't feel like this explicit enumeration will scale to a bunch more languages, and already involves a fair bit of duplication.  Can we just generate the various languages and variants through some declarative arrays and looping to process them?